godot-core/src/builtin/dictionary.rs

```diff
@@ -441,6 +447,15 @@ impl<'a> DictionaryIter<'a> {
         Some((key, value))
     }
 
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        let (remaining, overflows) = self.dictionary.len().overflowing_sub(self.next_idx);
+        if overflows {
+            return (0, Some(0));
+        }
+
+        (remaining, Some(remaining))
+    }
```

Comment:
You could just use [`saturating_sub`](https://doc.rust-lang.org/std/primitive.usize.html#method.saturating_sub) here, no?

Might be a good case to use fully qualified notation:
```rs
fn size_hint(&self) -> (usize, Option<usize>) {
    let remaining = usize::saturating_sub(self.dictionary.len(), self.next_idx);

    (remaining, Some(remaining))
}
```

Please also add a brief comment why simple subtraction is not sufficient 🙂

Reply:
Oh, that one I didn't know :)
Now it is in the latest version.